### PR TITLE
Add support for pinging healthchecks.io

### PR DESCRIPTION
**Pull request by tobiasmcnulty**
Adds optional support for pinging healthchecks.io from the main listen loop. Depends on the [healthchecks-io](https://github.com/andrewthetechie/py-healthchecks.io) Python package.

### CODE REVIEW

File: src/smpp_gateway/monitoring.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        if uuid:
+            self.ping_kwargs = {"uuid": uuid}
+        else:
+            self.ping_kwargs = {"slug": slug}
+        threading.Thread(target=self._worker_thread, daemon=True).start()
+
+    def _worker_thread(self):
+        from healthchecks_io.client.exceptions import HCAPIError
+
+        last_success_time = 0
```
Comment (Louis-Cloud) [inline src/smpp_gateway/monitoring.py:33]: Nice fun logic!

File: src/smpp_gateway/management/commands/smpp_client.py [hunk 2]
```diff
@@ -62,6 +62,21 @@ def add_arguments(self, parser):
...
+        )
+        parser.add_argument(
+            "--hc-ping-key",
+            default=os.environ.get("HEALTHCHECKS_IO_PING_KEY"),
+            help="Pings healthchecks.io with the specified ping key and check slug.",
+        )
+        parser.add_argument(
+            "--hc-check-slug",
+            default=os.environ.get("HEALTHCHECKS_IO_CHECK_SLUG"),
+            help="Pings healthchecks.io with the specified ping key and check slug.",
```
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply src/smpp_gateway/management/commands/smpp_client.py:78]: Good question! Both or neither of these arguments are required. I.e., a health check can be pinged with _either_ A) a UUID, or B) a ping key and a check slug. Can you think of a way to make this more clear?

Further reading: https://blog.healthchecks.io/2021/09/new-feature-slug-urls/

[the remark this answers is not in the document]
Comment (Louis-Cloud) [reply src/smpp_gateway/management/commands/smpp_client.py:78]: Oh I see! That makes sense!

Perhaps something like "Pings healthchecks.io with the specified ping key. (Need to include check slug arg as well)" and vice versa?

**Review by Louis-Cloud — APPROVED** [review]

File: src/smpp_gateway/management/commands/smpp_client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobiasmcnulty) [reply src/smpp_gateway/management/commands/smpp_client.py:78]: I like that! I updated as follows:

```
  --hc-check-uuid HC_CHECK_UUID
                        Pings healthchecks.io with the specified check UUID. If set, --hc-ping-key and --hc-check-slug will be
                        ignored.
  --hc-ping-key HC_PING_KEY
                        Pings healthchecks.io with the specified ping key and check slug. If set, --hc-check-slug must also be set.
  --hc-check-slug HC_CHECK_SLUG
                        Pings healthchecks.io with the specified ping key and check slug. If set, --hc-ping-key must also be set.
```

**Pull request merged**